### ISSUE TITLE
Setting content on the media player

### ISSUE THREAD

**Issue by ghost** [media-explorer/media-explorer#29]
Setting content on the media player should cause it to show it's self.

e.g. using SetUri from dbus/other methods

**Comment by ghost** [comment]
I'm thinking that something like this would do:

``` diff
diff --git a/mex/mex-media-dbus-bridge.c b/mex/mex-media-dbus-bridge.c
index 66b67a6..8c9c98d 100644
--- a/mex/mex-media-dbus-bridge.c
+++ b/mex/mex-media-dbus-bridge.c
@@ -24,6 +24,11 @@
 #include <dbus/dbus-glib-bindings.h>

 #include <mex/mex-player-common.h>
+#include <mex/mex-player.h>
+#include <mex/mex-feed.h>
+#include <mex/mex-program.h>
+#include <mex/mex-content.h>
+#include <mex/mex-content-view.h>

 static void mex_media_player_iface_init (MexMediaPlayerIface *iface);
 G_DEFINE_TYPE_WITH_CODE (MexMediaDBUSBridge,
@@ -145,10 +150,45 @@ mex_media_dbus_bridge_set_uri (MexMediaPlayerIface   *player_iface,
                                const gchar           *uri,
                                DBusGMethodInvocation *context)
 {
-  MexMediaDBUSBridge *bridge = MEX_MEDIA_DBUS_BRIDGE (player_iface);
-  MexMediaDBUSBridgePrivate *priv = bridge->priv;
+  MexPlayer *player;
+  gchar *mime_guess;
+  gchar *filename;
+  gchar *basename;
+
+  player = mex_player_get_default ();
+
+  MexFeed *fake_feed = mex_feed_new ("external", "external");
+  MexProgram *program = mex_program_new (fake_feed);
+
+  mex_content_set_metadata (MEX_CONTENT (program),
+                            MEX_CONTENT_METADATA_STREAM, uri);
+
+  mex_content_set_metadata (MEX_CONTENT (program),
+                            MEX_CONTENT_METADATA_URL, uri);
+
+  /* Try and scrape some metadata from the uri */
+  mime_guess = g_content_type_guess (uri, NULL, 0, NULL);
+
+  mex_content_set_metadata (MEX_CONTENT (program),
+                            MEX_CONTENT_METADATA_MIMETYPE, mime_guess);
+  g_free (mime_guess);
+
+  filename = g_filename_from_uri (uri, NULL, NULL);
+  basename = g_filename_display_basename (filename);
+
+  g_free (filename);
+
+  mex_content_set_metadata (MEX_CONTENT (program),
+                            MEX_CONTENT_METADATA_TITLE, basename);
+
+  g_free (basename);
+
+  mex_content_set_metadata (MEX_CONTENT (program),
+                            MEX_CONTENT_METADATA_SYNOPSIS,
+                            uri);

-  clutter_media_set_uri (priv->media, uri);
+  mex_content_view_set_content (MEX_CONTENT_VIEW (player),
+                                MEX_CONTENT (program));

   mex_media_player_iface_return_from_set_uri (context);
 }
diff --git a/mex/mex-player.c b/mex/mex-player.c
index 7c4d7d5..3d272ee 100644
--- a/mex/mex-player.c
+++ b/mex/mex-player.c
@@ -95,6 +95,7 @@ struct _MexPlayerPrivate
 enum
 {
   CLOSE_REQUEST,
+  OPEN_REQUEST,

   LAST_SIGNAL
 };
@@ -155,6 +156,28 @@ mex_player_set_content (MexContentView *view,
 {
   MexPlayerPrivate *priv = MEX_PLAYER (view)->priv;

+  /* If the content we're getting is being set from external source such
+     as dbus then we need to ask the main app to display the player */
+    if (MEX_IS_PROGRAM (content))
+      {
+        MexFeed *feed;
+        const gchar *source;
+
+        feed = mex_program_get_feed (MEX_PROGRAM (content));
+
+        g_object_get (feed, "source", &source, NULL);
+
+        if (g_strcmp0 (source, "external") == 0)
+          {
+            g_signal_emit (MEX_PLAYER (view), signals[OPEN_REQUEST], 0);
+
+            mex_media_controls_set_content (MEX_MEDIA_CONTROLS (priv->controls),
+                                            content,
+                                            MEX_MODEL (feed));
+          }
+      }
+
+
   if (priv->model)
     mex_media_controls_set_content (MEX_MEDIA_CONTROLS (priv->controls),
                                     content, priv->model);
@@ -550,6 +573,13 @@ mex_player_class_init (MexPlayerClass *klass)
                                          g_cclosure_marshal_VOID__VOID,
                                          G_TYPE_NONE, 0);

+  signals[OPEN_REQUEST] = g_signal_new ("open-request",
+                                        G_TYPE_FROM_CLASS (klass),
+                                        G_SIGNAL_RUN_LAST,
+                                        0, NULL, NULL,
+                                        g_cclosure_marshal_VOID__VOID,
+                                        G_TYPE_NONE, 0);
+
   pspec = g_param_spec_boolean ("idle-mode",
                                 "Idle Mode",
                                 "Idle Mode",
diff --git a/shell/mex-main.c b/shell/mex-main.c
index ad2a199..fb358b6 100644
--- a/shell/mex-main.c
+++ b/shell/mex-main.c
@@ -341,6 +341,17 @@ mex_play_from_begin_cb (MxAction *action, MexData *data)
 }

 static void
+mex_player_content_set_externally_cb (MexData *data)
+{
+  /* hide other actors */
+  mex_hide_actor (data, data->explorer);
+  mex_hide_actor (data, data->slide_show);
+
+  /* show the video player */
+  mex_show_actor (data, data->player);
+}
+
+static void
 mex_show_cb (MxAction *action, MexData *data)
 {
   MexContent *content = mex_action_get_content (action);
@@ -356,7 +367,7 @@ mex_show_cb (MxAction *action, MexData *data)
   /* start the slide show */
   mex_content_view_set_context (MEX_CONTENT_VIEW (data->slide_show), model);

-  /* hide other actors */
+ /* hide other actors */
   mex_hide_actor (data, data->player);

   /* stop any playing video, even in idle mode */
@@ -2474,6 +2485,10 @@ main (int argc, char **argv)
   data.player = (ClutterActor *) mex_player_get_default ();
   g_signal_connect_swapped (data.player, "close-request",
                             G_CALLBACK (mex_go_back), &data);
+  g_signal_connect_swapped (data.player, "open-request",
+                            G_CALLBACK (mex_player_content_set_externally_cb),
+                            &data);
+
   clutter_container_add_actor (CLUTTER_CONTAINER (data.stack),
                                data.player);
```

**Comment by ghost** [comment]
Implemented in slightly different way: 6e17edf8f55051eccca225bb9d7e5c4f4e013325
